### ISSUE TITLE
Aria-label getting rewritten in DropdownMenu

### ISSUE THREAD

**Issue by mmu-adamnicolson** [foundation/foundation-sites#12447]
## What should happen?
When you produce a link in a DropdownMenu that contains a valid aria-label, please can you make it so that its not re-wrriten with the content of the link itself. 
...

## What happens instead?
Your framework (Nest I think may be the culprit, version 6.6.3 doesn't do the same thing), replaces my valid arial-labels and replaces it with one that is generated based on the link's text.
...

## Possible Solution

Detect IF the link does not have an aria-label, and continues to generate if you really need to (not actually helping anyone IMO). Don't rewrite our aria-labels with yours. 
...

## Test Case and/or Steps to Reproduce (for bugs)

Test Case: https://codepen.io/adamnicolson/pen/MWVaVmy Once the menu has been initialised, look at the aria-label you are producing.
If you roll the version back to 6.6.3 the same behaviour cannot be seen.

How to reproduce:
1. Add an aria-label to a link inside a DropdownMenu.
2. Load the page rendered HTML. The arai-label will be replaced with a non-helpful generic version based on the link-text. 

## Context

...

## Your Environment

- Foundation version(s) used: latest
- Browser(s) name and version(s): all
- Device, Operating System and version: 
- Link to your project: 

## Checklist

- [x] I have read and follow the CONTRIBUTING.md document.
- [x] There are no other issues similar to this one.
- [x] The issue title and template are correctly filled.

**Comment by mmu-adamnicolson** [comment]
Seems to have happened after 6.6.3. The issue can be seen on 6.7.0
